pyqgiswps/executors/io/filesio.py

```diff
@@ -128,18 +129,35 @@ def parse_output_definition(outdef: QgsProcessingOutputDefinition, kwargs,
     elif isinstance(outdef, QgsProcessingOutputFile):
         # Try to get a corresponding inputFileDefinition
         # See https://qgis.org/pyqgis/master/core/QgsProcessingParameterFileDestination.html
-        mime = None
+        supported_formats = []
         if alg:
             inputdef = alg.parameterDefinition(outdef.name())
             if isinstance(inputdef, QgsProcessingParameterFileDestination):
-                mime = mimetypes.types_map.get("." + inputdef.defaultFileExtension())
+                default_extension = inputdef.defaultFileExtension()
+                default_extension_idx = -1
+                for filter_ in inputdef.fileFilter().split(";;"):
+                    extension_regex = QRegularExpression("[*.]([a-z]+)")
+                    match = extension_regex.match(filter_)
+                    if match.hasMatch():
+                        extension = match.captured(1)
+                        if extension == default_extension:
+                            default_extension_idx = len(supported_formats)
+
+                        supported_formats.append(Format(match.captured(1)))
+
+                # The first format in the list is interpreted as the default one
+                # Ensure that it corresponds to `default_extension`
+                if default_extension_idx > 0:
+                    supported_formats.insert(0, supported_formats.pop(Format(default_extension)))
```

Comment:
This is not OGC compliant: the Format argument must be a valid mime type not a file extension.  You must use the mimetypes module to resolve file extension to valide mime types.

See https://github.com/3liz/py-qgis-wps/blob/master/pyqgiswps/inout/formats.py#L35

Reply:
My bas. This is fixed now.